daemon/station_reporter.rs

```diff
@@ -0,0 +1,117 @@
+use std::cell::RefCell;
+use std::io::{stderr, stdout, Write};
+use std::time::Duration;
+
+use serde_json::{json, Map};
+use zinnia_runtime::anyhow::Result;
+use zinnia_runtime::{JobCompletionTracker, LogLevel, Reporter};
+
+/// StationReporter reports activities to stdout as ND-JSON stream and all Console logs to stderr
+pub struct StationReporter {
+    tracker: RefCell<JobCompletionTracker>,
+    module_name: String,
+    log_target: String,
+}
+
+impl StationReporter {
+    /// Create a new instance.
+    ///
+    /// `job_report_delay` specifies how often the information about new jobs is printed.
+    pub fn new(job_report_delay: Duration, module_name: String) -> Self {
+        let log_target = format!("module:{module_name}");
+        Self {
+            tracker: RefCell::new(JobCompletionTracker::new(job_report_delay)),
+            module_name,
+            log_target,
+        }
+    }
+
+    fn print_jobs_completed(&self, total: u64) {
+        // TODO: print data from all modules
+        // https://github.com/filecoin-station/zinnia/issues/144
+        // modules: {"saturn": 100, "retrieval-checker": 23}}
+        let mut modules = Map::new();
+        modules.insert(self.module_name.clone(), json!(total));
+
+        let event = json!({
+            "type": "jobs-completed",
+            "total": total,
+             "modules": modules,
+        });
+
+        let _ = print_event(&event);
```

Comment:
```suggestion
        print_event(&event);
```

Would this work too?